### ISSUE TITLE
New build of Boost

### ISSUE THREAD

**Issue by jakirkham** [conda/conda-recipes#579]
Any chance we could get this rebuilt? In particular, it would be really nice to see Python 2 and Python 3 packages of Boost to get Boost.Python working for these respective Python environments. This appears to be supported by the current recipe here.

**Comment by msarahan** [comment]
Yes.  This is a dependency for something we've been requested to build (Caffe) by a paying customer, so they will be updated soon.  Any platform in particular?  I ask because it looks like Windows is going to take a while, with some packages requiring brand new build files (perhaps CMake) for lack of Windows support.

**Comment by ccordoba12** [comment]
@msarahan, the recipes present here compile on Windows too for all our supported Python versions (that was my main improvement a couple of months ago ;-)

Could you create a PR against our internal repo with them or do you want me to do it?

**Comment by jakirkham** [comment]
Oh ok, if it works on all platforms that is great and will definitely be useful. ATM, I was looking for Mac and Linux packages.

**Comment by msarahan** [comment]
@ccordoba12 if you have time, you'll get there faster than me.  I am swamped trying to get conda PR 1727 wrapped up.

**Comment by ukoethe** [comment]
Does the Windows build include the `boost_python` library? IIRC, it used to be necessary to explicitly specify `--with-python` in the build script command line, which is missing in `boost/bld.bat`. Python support would be important for me.

**Comment by jakirkham** [comment]
Yes, it appears this is true of the recipe on all platforms.

**Comment by ccordoba12** [comment]
@ukoethe, not right now but it will once our packages are updated with the recipes present here.

**Comment by jakirkham** [comment]
Could I ask how we got this line ( https://github.com/conda/conda-recipes/commit/af365755193a3a42c3a56b84e09649a438be32ce ) and why it is necessary?

**Comment by msarahan** [comment]
How do we know we use GCC44 specifically?  Or why that needs to be fed into
the other file?  GCC44 is the standard version of GCC we use right now.  It
was the latest version provided in CentOS 5.  There's a Red Hat download
that might be useful to get us to GCC 4.8.  We're working on examining
that.  For now, GCC 4.4 is what we have.

**Comment by jakirkham** [comment]
So, I am raising this for a few reasons.

First, I also use CentOS (though 6.3) and it does have a default `gcc` (4.4.7), but there is no `gcc44` or `g++44`. As a result, when I try to build this it looks for that copy of `g++44` and then can't find it and fails. Is it not possible to use the normal `g++` here?

Second, normally I don't build with the GNU 4.4.7 compilers as they are too old for some of my codebase which requires C++11 features. Instead I install a copy of GNU 4.8 compilers and add them to environment. Hardcoding paths and versions hurts this use case. Would it be possible to not specify this at all?

**Comment by ccordoba12** [comment]
Well, CentOS 5 and gcc 4.4 are our standards for building things on Linux, and that's reflected on the recipe present here. And standards are standards, so that's not going to change.

But you're welcome to fork our recipe and adapt it to your own purposes :-)

**Comment by ukoethe** [comment]
> Well, CentOS 5 and gcc 4.4 are our standards for building things on Linux, and that's reflected on the recipe present here. And standards are standards, so that's not going to change.

conda should be able to support multiple compilers on all platforms. In fact, the version solver overhaul in https://github.com/conda/conda/pull/2000 aims (among other things) at exactly this: by specifying `features` and `track_features`, multiple build types can coexist on anaconda, and `conda install` will take care of selecting matching binaries. Once this PR is ready, you should definitely generalize your recipes, instead of advising users to fork if they can't live with an outdated "standard".

**Comment by ccordoba12** [comment]
> instead of advising users to fork if they can't live with an outdated "standard".

I'm not saying exactly that. These are just our recipes, i.e. the recipes we maintain and use to create our own packages. If you want to build different packages (i.e. with different gcc's and environments), then you can fork them, that's all :-) These are not canonical or set on stone things, in the same way as there's no canonical Linux distribution, for example.

Sure when `track_features` is better implemented we can switch to that, but in the meantime, what's the problem with forking? We are not forcing @jakirkham or anyone to use the recipes present here :-)

**Comment by jakirkham** [comment]
It certainly isn't blocking me from building them as I just ripped out that line, but I figured I would mention it here because it seems like this could be improved IMHO.

**Comment by msarahan** [comment]
The nicest fix on our end is probably to rip that out, and force gcc to be a symlink to gcc44 on our build machines.  I'm not sure if/why that isn't done already.  I'll ask.  Hardcoding versions like this just makes more work for everyone.

**Comment by ukoethe** [comment]
> These are just our recipes, i.e. the recipes we maintain and use to create our own packages.

Ah, I naively assumed that https://github.com/conda/conda-recipes, by being part of the conda account, is somewhat more official than other recipe collections. :-) Glad to hear that you might upgrade once `track_features` works properly.

**Comment by jakirkham** [comment]
> Hardcoding versions like this just makes more work for everyone.

For sure. I figured I'd mention it since you were thinking about doing a build anyways. :)

In any event, it is probably low priority.

**Comment by jakirkham** [comment]
Something else to think about with regards to the compiler. As `conda` provides a pretty good copy of `gcc`, maybe that should be used instead of something that is included with the VM.

**Comment by ccordoba12** [comment]
New builds are (finally) coming, hopefully by the end of the week :-)

I also removed the line that made Boost compile against gcc 4.4.

**Comment by jakirkham** [comment]
Sounds good. Thanks for the update. Feel free to ping me again when they roll out.

**Comment by jakirkham** [comment]
Awesome! Thanks for the update.
